src/lib/runtime/mounts/dev/dev.c

```diff
@@ -103,6 +105,17 @@ int _singularity_runtime_mount_dev(void) {
         bind_dev(sessiondir, "/dev/random");
         bind_dev(sessiondir, "/dev/urandom");
 
+        /* if the user passed the --nv flag and the --contain flag, still bind
+        nvidia devices */
+        if ( nvdevs != NULL ) {
+            char *nvdev = strtok(nvdevs, ",");
+            while ( nvdev != NULL ) {
+                singularity_message(2, "Binding device %s\n", nvdev);
+                bind_dev(sessiondir, nvdev);
```

Comment:
@GodloveD That's introduce a security issue, SINGULARITY_NVDEV is fully controlled by user so he could bind anything he want into container without sanity checks